addon/mixins/serializer.js

```diff
@@ -0,0 +1,32 @@
+import Mixin from '@ember/object/mixin'
+import { inject as service } from '@ember/service'
+
+export default Mixin.create({
+  /**
+   * Services
+   */
+
+  recordMetaService: service('record-meta'),
+
+  /**
+   * Hooks
+   */
+
+  normalize () {
+    this.normalizeRecordMeta(...arguments)
+
+    return this._super(...arguments)
+  },
+
+  normalizeRecordMeta (modelClass, payload) {
```

Comment:
This can easily be split up into a separate pure utility right?

I would just let the user manually inject the service, and pass the data to that util function. This would remove the need for the mixin, but you could still provide it for those that don't mind using them?

Reply:
Okay, I'll add the util (I'll also document this as the recommended approach then).

Reply:
This would also immediately "solve" the casing issues I guess? Going for the util approach the user would still have the opportunity to modify the meta object. E.g. camelize all keys or something.